### PR TITLE
avoid async overhead in ReadNextLineAsync

### PR DESCRIPTION
**Pull request by geoffkizer**
Updated version of #23210 

@stephentoub 

I'm blocked on getting perf data here for now, but I expect it to be similar to the previous PR.

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs [hunk 1]
```diff
@@ -814,47 +814,59 @@ private Task WriteToStreamAsync(byte[] buffer, int offset, int count, Cancellati
...
+        private async ValueTask<ArraySegment<byte>> ReadNextLineSlowAsync(CancellationToken cancellationToken)
         {
-            int searchOffset = 0;
             while (true)
             {
-                int startIndex = _readOffset + searchOffset;
-                int length = _readLength - startIndex;
-                int crPos = Array.IndexOf(_readBuffer, (byte)'\r', startIndex, length);
-                if (crPos < 0)
+                await FillAsync(cancellationToken);
```
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs]: .ConfigureAwait(false)

File: src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs [hunk 2]
```diff
@@ -864,6 +876,29 @@ private Task WriteToStreamAsync(byte[] buffer, int offset, int count, Cancellati
...
 
             _readOffset += 2;
+            return true;
+        }
+
+        private async Task ReadCrLfSlowAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                await FillAsync(cancellationToken);
```
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs]: .ConfigureAwait(false)

**Comment by karelz** [comment]
@geoffkizer what is the status of this PR? It's almost 3 weeks old now ...

**Comment by karelz** [comment]
@geoffkizer ping?

**Comment by karelz** [comment]
@geoffkizer any update?

**Comment by stephentoub** [comment]
@geoffkizer, did you want to move ahead with this PR?  Or should we close it for now?

**Comment by karelz** [comment]
Closing for now. We can reopen later, when it is ready.

**Pull request closed without merging**